fe/fe-core/src/main/java/com/starrocks/catalog/HudiTable.java

```diff
@@ -0,0 +1,391 @@
+        return partColumnNames;
+    }
+
+    public List<String> getDataColumnNames() {
+        return dataColumnNames;
+    }
+
+    public Map<PartitionKey, Long> getPartitionKeys() throws DdlException {
+        List<Column> partColumns = getPartitionColumns();
+        return Catalog.getCurrentCatalog().getHiveRepository()
+                .getPartitionKeys(resourceName, db, table, partColumns);
+    }
+
+    public String getResourceName() {
+        return resourceName;
+    }
+
+    public HoodieTableType getTableType() {
+        return HoodieTableType.valueOf(hudiProperties.get(HUDI_TABLE_TYPE));
+    }
+
+    public String getHudiHiveMetastoreUris() {
+        return hudiProperties.get(HUDI_METASTORE_URIS);
+    }
+
+    public String getHudiBasePath() {
+        return hudiProperties.get(HUDI_BASE_PATH);
+    }
+
+    public String getHudiTableInputFormat() {
+        return hudiProperties.get(HUDI_TABLE_INPUT_FORMAT);
+    }
+
+    public String getHudiMatadataEnable() {
+        return hudiProperties.get(HUDI_MATADATA_ENABLE);
+    }
+
+    private void validate(Map<String, String> properties) throws DdlException {
+        if (properties == null) {
+            throw new DdlException("Please set properties of iceberg table, they are: database, table.");
+        }
+
+        Map<String, String> copiedProps = Maps.newHashMap(properties);
+        this.db = copiedProps.get(HUDI_DB);
+        if (Strings.isNullOrEmpty(db)) {
+            throw new DdlException(String.format(PROPERTY_MISSING_MSG, HUDI_DB, HUDI_DB));
+        }
+        copiedProps.remove(HUDI_DB);
+
+        this.table = copiedProps.get(HUDI_TABLE);
+        if (Strings.isNullOrEmpty(table)) {
+            throw new DdlException(String.format(PROPERTY_MISSING_MSG, HUDI_TABLE, HUDI_TABLE));
+        }
+        copiedProps.remove(HUDI_TABLE);
+
+        String hudiMetaDataEnable = copiedProps.get(HUDI_MATADATA_ENABLE);
+        if (!Strings.isNullOrEmpty(hudiMetaDataEnable)) {
+            copiedProps.remove(HUDI_MATADATA_ENABLE);
+            hudiProperties.put(HUDI_MATADATA_ENABLE, hudiMetaDataEnable);
+        } else {
+            hudiProperties.put(HUDI_MATADATA_ENABLE, "false");
+        }
+
+        String resourceName = copiedProps.get(HUDI_RESOURCE);
+        Resource resource = Catalog.getCurrentCatalog().getResourceMgr().getResource(resourceName);
+        HudiResource hudiResource = (HudiResource) resource;
+        if (hudiResource == null) {
+            throw new DdlException("hudi resource [" + resourceName + "] not exists");
+        }
+        if (hudiResource.getType() != Resource.ResourceType.HUDI) {
+            throw new DdlException("resource [" + resourceName + "] is not hudi resource");
+        }
+        hudiProperties.put(HUDI_METASTORE_URIS, hudiResource.getHiveMetastoreURIs());
+
+        org.apache.hadoop.hive.metastore.api.Table metastoreTable = Catalog.getCurrentCatalog().getHiveRepository()
```

Comment:
Does this mean that hudi only use hms to store table?

Reply:
no, hudi work well without HMS.